macros/sql/get_column_values.sql

```diff
@@ -36,16 +37,26 @@ Returns:
 
         {%- else -%}
 
-            select
-                {{ column }} as value
-
-            from {{ target_relation }}
-            group by 1
-            order by count(*) desc
-
-            {% if max_records is not none %}
-            limit {{ max_records }}
-            {% endif %}
+            with sorted_column_values as (
+
+                select
+                    {{ column }} as value,
+                    {%- if sort_column is none %}
+                    count(*) as sort_column
+                    {% else %}
+                    {# We take the max sort value for each value to make sure 
+                        there are no duplicate rows for each value #}
+                    max({{ sort_column }}) as sort_column
+                    {% endif %}
+                from {{ target_relation }}
+                group by 1
+                order by 2 {{ sort_direction if sort_direction else "desc" }}
```

Comment:
If we go with the above comment, it may be easier to do something like this:
```suggestion
                order by {{ order_by }} {{ sort_direction or "desc" }}
```

Reply:
Still trying to figure out how to make this work with non-aggregate columns. E.g. you pass in `column_2` instead `count(*)` we need to wrap in an aggregation, like `max(column_2)`, or group by it as well.